src/app/store/technical-records/effects/technical-record-service.effects.ts

```diff
@@ -158,4 +170,25 @@ export class TechnicalRecordServiceEffects {
         return `${this.apiErrors[type + '_400']} ${search ? search : JSON.stringify(error.error)}`;
     }
   }
+
+  generateTechRecordBasedOnSectionTemplates = createEffect(() =>
+    this.actions$.pipe(
+      ofType(changeVehicleType),
+      mergeMap(action => of(action).pipe(withLatestFrom(this.store.pipe(select(editableTechRecord))), take(1))),
+      concatMap(([action, editableTechRecord]) => {

```

Comment:
You can flatten this out to omit the nesting:
 ```suggestion
      ofType(changeVehicleType),
      withLatestFrom(this.store.pipe(select(editableTechRecord)),
      take(1),
      concatMap(([action, editableTechRecord]) => {
```

Reply:
This should work, but depending on your use case, you might need to insert the `take(1)` into the store's pipe.
```typescript
withLatestFrom(this.store.pipe(select(editableTechRecord), take(1)),
```

Reply:
On it